### ISSUE TITLE
Make a better UI for charts

### ISSUE THREAD

**Issue by lusebille** [gristlabs/grist-core#1094]

**Comment by manuhabitela** [comment]
:wave: 

I agree!

- would be great to think about color contrast ratios. The best would be to ensure a ratio of 4.5:1 when showing text on colored parts. Here in the screenshots for example the almost-black + orange and white on green have good contrast if we are sure the text is large enough (the WCAG rule is 3:1 ratio in case of >24px text), but for now those texts can be rendered pretty small.
- Great thing now is each color are pretty contrasted between them, _and_ each part has a label on top or next to it. If keeping that logic, I'd say adding a visible border between each part to ensure we always understand each part limit would be great. Like this for example (before/after):

![image](https://github.com/user-attachments/assets/c8be8e02-fd90-4862-9efe-209343e19a9d)

![image](https://github.com/user-attachments/assets/2cb71dac-ef66-4007-b1bc-45bcb9608056)

- another way of making sure every part of the chart is understood would be to stop relying exclusively on colors, like this chart. In case of colors not contrasted enough next to each other, we ensure the difference can be understood anyway:

![image](https://github.com/user-attachments/assets/8da30c86-f15f-476f-a0cd-82678cce8dd8)
